File: studijni-materialy/05-funkce/06-priklady/main.js
let multiplier = 3;

let a = 2;
let b = 3;

console.log(sum(a, b));

console.log(multiply(a));

changeMultiplier(4);

console.log(multiply(a));

function sum(a, b) {
    let result = a + b;
    return result;
}

function multiply(number) {
    // přístup k vnější proměnné `mutliplier`
    let result = number * multiplier;
    return result;
}

function changeMultiplier(newMultiplier) {
    // změna vnější proměnné `mutliplier`
    multiplier = newMultiplier;
}

File: studijni-materialy/05-funkce/06-priklady/main.cs
using System;

internal class Program
{
    static int multiplier = 3;

    static void Main(string[] args)
    {
        int a = 2;
        int b = 3;

        Console.WriteLine(Sum(a, b));

        Console.WriteLine(Multiply(a));

        ChangeMultiplier(4);

        Console.WriteLine(Multiply(a));
    }

    static int Sum(int a, int b)
    {
        int sum = a + b;
        return sum;
    }

    static int Multiply(int number)
    {
        // přístup k vnější proměnné `mutliplier`
        int result = number * multiplier;
        return result;
    }

    static void ChangeMultiplier(int newMultiplier)
    {
        // změna vnější proměnné `mutliplier`
        multiplier = newMultiplier;
    }
}

